runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -339,7 +339,8 @@ private void prepareStateBackend(K key) {
           // Key for state request is shipped already encoded as ByteString,
           // this is mostly a wrapping with ByteBuffer. We still follow the
           // usual key encoding procedure.
-          final ByteBuffer encodedKey = FlinkKeyUtils.encodeKey(key, keyCoder);
+          // final ByteBuffer encodedKey = FlinkKeyUtils.encodeKey(key, keyCoder);
+          final ByteBuffer encodedKey = ByteBuffer.wrap(key.toByteArray());
```

Comment:
The ByteBuffer wrapping is an implementation detail of `FlinkKeyUtils.encodeKey`.